internal/mode/static/manager.go

```diff
@@ -67,6 +69,11 @@ func StartManager(cfg config.Config) error {
 		return fmt.Errorf("cannot build runtime manager: %w", err)
 	}
 
+	controlConfigNSName := types.NamespacedName{
+		Namespace: cfg.Namespace,
+		Name:      cfg.ConfigCRDName,
+	}
+
 	// Note: for any new object type or a change to the existing one,
 	// make sure to also update prepareFirstEventBatchPreparerArgs()
```

Comment:
could you update prepareFirstEventBatchPreparerArgs() too ?

Reply:
Isn't the purpose of this to ensure we build the full nginx config before first publish? We shouldn't have to include this config in that batch, right?

Reply:
yep. it was introduced because of that.

let consider the case when the NKG starts and the config exists. With the current logic, the handler will process NginxConfig at some point after the start. This mean NKG will run for some time with the default values, rather than configured values.

I think my suggestion doesn't fix that either - because processing the first batch starts at some point after a lot of NKG components already has started.

So I wonder now if it makes sense to fetch the config before starting all the controllers and the manager?

Reply:
Yeah that's definitely an issue, but I believe we have to start the manager in order for our client to be able to access the cache of objects that exist. The only way that I'm aware of is if we initialize a separate client directly from `client-go`. Even then, that won't happen immediately.

How important is it that the configuration is applied while we are still "starting up"?

Reply:
what about this method of the Manager?
```go
	// GetAPIReader returns a reader that will be configured to use the API server.
	// This should be used sparingly and only when the client does not fit your
	// use case.
	GetAPIReader() client.Reader
```

> How important is it that the configuration is applied while we are still "starting up"?

I think it is best to read it before we start all the components, so that we can enable desired log level for all components.